ui/src/components/WorkflowVariablepicker/index.tsx

```diff
@@ -0,0 +1,173 @@
+import { Fragment, useState } from "react";
+import {
+  Variablepicker,
+  VariablepickerError,
+  VariablepickerHeading,
+  VariablepickerItem,
+  VariablepickerMessage,
+  VariablepickerSeparator,
+} from "~/design/VariablePicker";
+
+import { ButtonBar } from "~/design/ButtonBar";
+import Input from "~/design/Input";
+import { VarSchema } from "~/api/variables/schema";
+import { useNamespace } from "~/util/store/namespace";
+import { useTranslation } from "react-i18next";
+import { useWorkflowVariables } from "~/api/tree/query/variables";
+import { z } from "zod";
+
+const variableWithoutChecksumAndSize = VarSchema.omit({
+  checksum: true,
+  size: true,
+});
+
+type variableType = z.infer<typeof variableWithoutChecksumAndSize>;
+
+const WorkflowVariablePicker = ({
+  namespace: givenNamespace,
+  workflow,
+  defaultVariable,
+  onChange,
+}: {
+  namespace?: string;
+  workflow: string;
+  defaultVariable?: string;
+  onChange: (variable: variableType | undefined) => void;
```

Comment:
Yes, in the design phase I thought it was a good idea with a new empty variable object, because I assumed we also need other properties. Next time I will check first where the component will be implemented in the end. :)